src/convert.rs

```diff
@@ -0,0 +1,379 @@
+//! A converter to convert between the Garble Circuit and the Bristol fashion circuit format.
+//! https://nigelsmart.github.io/MPC-Circuits/
+
+use crate::circuit::{Circuit, Gate, PANIC_RESULT_SIZE_IN_BITS};
+use std::{
+    collections::HashSet,
+    fs::File,
+    io::{BufRead, BufReader, Write},
+};
+
+/// An error that occurred during compilation.
+#[derive(Debug)]
+pub enum ConverterError {
+    /// Some input wires are also output wires.
+    ToBristolError(ToBristolError),
+    /// An error occurred while writing to the file.
+    FromBristolError(FromBristolError),
+    /// An error occurred while writing to the file.
+    IoError(std::io::Error),
+}
+
+/// An error that occurred during compilation.
+#[derive(Debug)]
+pub enum ToBristolError {
+    /// Some input wires are also output wires.
+    OutputWireIsInput,
+}
+
+/// An error that occurred during compilation.
+#[derive(Debug)]
+pub enum FromBristolError {
+    /// An unknown gate was encountered.
+    UnknownGate(String),
+    /// Missing number of inputs in the gate definition.
+    MissingNumberOfInputs,
+    /// Missing input wires in the gate definition.
+    MissingInputWires,
+    /// Missing output wires in the gate definition.
+    MissingOutputWires,
+    /// Missing gate type in the gate definition.
+    MissingGateType,
+    /// An error occurred while parsing the file.
+    OtherParseError(String),
+    /// The number of input wires does not match the number of parties.
+    InputPartiesMismatch(usize, usize),
+    /// The number of output wires does not match the number of output gates.
+    OutputCountMismatch(usize, usize),
+    /// Missing line in the Bristol fashion circuit file.
+    MissingLine,
+    /// Malformed Bristol fashion circuit file.
+    MalformedLine(String),
+}
```

Comment:
I think these should probably be `#[non_exhaustive]`, so we can add error variants later without a breaking change.